payjoin/src/uri/error.rs

```diff
@@ -7,10 +9,27 @@ pub(crate) enum InternalPjParseError {
     DuplicateParams(&'static str),
     MissingEndpoint,
     NotUtf8,
-    BadEndpoint,
+    BadEndpoint(BadEndpointError),
     UnsecureEndpoint,
 }
 
+#[derive(Debug)]
+pub enum BadEndpointError {
+    UrlParse(ParseError),
+    #[cfg(feature = "v2")]
+    LowercaseFragment,
+}
+
+impl std::fmt::Display for BadEndpointError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BadEndpointError::UrlParse(err) => write!(f, "Invalid URL: {:?}", err),
+            BadEndpointError::LowercaseFragment =>
```

Comment:
If it's feature gated above we need a gate here in order to build `v1` with `--no-default-features` -- which may be otherwise broken because we're only testing checks with `--all-features` in CI now, whoops!

```suggestion
            #[cfg(feature = "v2")]
            BadEndpointError::LowercaseFragment =>
```

Reply:
That's weird its not refreshing here,
https://github.com/payjoin/rust-payjoin/blob/b8e9768158ff1348a3972722d98c9f4e2e4f4363/payjoin/src/uri/error.rs#L27-L28

Reply:
I think I just spammed 2 rebases too quickly for github to make sense of it